src/main/java/teammates/logic/core/StudentsLogic.java

```diff
@@ -585,39 +585,30 @@ private StudentEnrollDetails enrollStudent(StudentAttributes validStudentAttribu
     public List<StudentAttributes> createStudents(String lines, String courseId) throws EnrollException {
         List<String> invalidityInfo = new ArrayList<>();
         String[] linesArray = lines.split(Const.EOL);
-        ArrayList<String> studentEmailList = new ArrayList<>();
         List<StudentAttributes> studentList = new ArrayList<>();
 
         StudentAttributesFactory saf = new StudentAttributesFactory(linesArray[0]);
 
         for (int i = 1; i < linesArray.length; i++) {
             String line = linesArray[i];
             String sanitizedLine = SanitizationHelper.sanitizeForHtml(line);
+            if (StringHelper.isWhiteSpace(line)) {
+                continue;
+            }
             try {
-                if (StringHelper.isWhiteSpace(line)) {
-                    continue;
-                }
                 StudentAttributes student = saf.makeStudent(line, courseId);
 
                 if (!student.isValid()) {
-                    String info = StringHelper.toString(SanitizationHelper.sanitizeForHtml(student.getInvalidityInfo()),
-                                                    "<br>" + Const.StatusMessages.ENROLL_LINES_PROBLEM_DETAIL_PREFIX + " ");
-                    invalidityInfo.add(String.format(Const.StatusMessages.ENROLL_LINES_PROBLEM, sanitizedLine, info));
+                    addInvalidStudentInfo(invalidityInfo, student, sanitizedLine);
                 }
-
-                if (isStudentEmailDuplicated(student.email, studentEmailList)) {
-                    String info =
-                            StringHelper.toString(
-                                    getInvalidityInfoInDuplicatedEmail(student.email, studentEmailList, linesArray),
-                                    "<br>" + Const.StatusMessages.ENROLL_LINES_PROBLEM_DETAIL_PREFIX + " ");
-                    invalidityInfo.add(String.format(Const.StatusMessages.ENROLL_LINES_PROBLEM, sanitizedLine, info));
+                int isDuplicate = getDuplicateIndex(student.email, studentList);
```

Comment:
Let's go with duplicateEmailIndex. 

The method should be renamed to getDuplicateEmailIndex for better clarity.